test/jdk/java/io/File/GetCanonicalPath.java

```diff
@@ -122,4 +125,96 @@ void driveLetter() throws IOException {
         String path = new File("c:/").getCanonicalPath();
         assertFalse(path.length() > 3, "Drive letter incorrectly represented");
     }
+
+    private static Path createPath(String pathname) throws IOException {
+        File file = new File(pathname);
+        file.deleteOnExit();
+        return file.toPath();
+    }
+
+    private static boolean testLinks = true;
```

Comment:
It might be clearer to replace this with supportsLinks and have the tests Assumptions.asserTrue(supportsLinks) so they will be skipped when not supported.